client-core/src/service/key_service.rs

```diff
@@ -56,7 +75,152 @@ where
 
     /// Clears all storage
     pub fn clear(&self) -> Result<()> {
-        self.storage.clear(KEYSPACE)
+        self.storage.clear(KEYSPACE).expect("clear keyspace");
+        self.storage
+            .clear(KEYSPACE_HD)
+            .expect("clear keyspace for hd");
+        Ok(())
+    }
+
+    /// get random mnemonic
+    pub fn get_random_mnemonic(&self) -> String {
+        let mnemonic = Mnemonic::new(MnemonicType::Words24, Language::English);
+        mnemonic.clone().to_string()
+    }
+
+    /// is hd
+    pub fn is_hd_wallet(&self, name: &str, passphrase: &SecUtf8) -> bool {
+        let key = name.as_bytes();
+        let value = self.read_value(passphrase, &key[..]);
+        value.is_some()
+    }
+    /// generate seed from mnemonic
+    pub fn generate_seed(&self, mnemonic: &str, name: &str, passphrase: &SecUtf8) -> Result<()> {
+        debug!("hdwallet generate seed={}", mnemonic);
+        let mnemonic = Mnemonic::from_phrase(&mnemonic.to_string(), Language::English).unwrap();
```

Comment:
From the above we are generating 24 mnemonic words by default, will we restrict input to be 24 words or we will accept importing any mnemonics that is valid bip39?

Reply:
yes, it will accept any bip39 standard mnemonics.
12 words, 15 words.. 24 words